### ISSUE TITLE
Question: mido.filename Not Set With Save to New File - True?

### ISSUE THREAD

**Issue by JeffRocchio** [mido/mido#589]
Using the docs @ [https://mido.readthedocs.io/en/stable/files/midi.html#creating-a-new-file](https://mido.readthedocs.io/en/stable/files/midi.html#creating-a-new-file) I:

- Create a new midi file object-instance. 
- I Append a track and write data to that. 
- Then I save the file to my hard drive. 
- If, immediately after, I query the newly saved mido.midifile object for it's filename: mido.MidiFile.filename I get the result "None."

I'm here just looking for a confirm that is correct behavior (i.e., I need to keep my user-supplied filename around for later use even after the midi file object save).

My actual code snips:

```python
class OutputMIDI:
    def __init__(self, outputFile):
        self.midiOut = mido.MidiFile()
        self.outFilePath = outputFile

    def timeshiftTrack(self, track):
        pass # <-- figure this out. I think this should be a private function that gets called by appendTrack().

    def appendTrack(self, track):
        self.midiOut.tracks.append(track)

    def writeFile(self):
        print(f'Writing File - outfile = {self.outFilePath}')
        self.midiOut.save(self.outFilePath)
        # Just test if the save updates the object's filename param:
        print(f'File Written - outfile = {self.midiOut.filename}')

```
After building the outputMIDI, when I call writeFile() I get:

```
Writing File - outfile = /home/jroc/Dropbox/projects/Audio_AddCountIn_MIDI/test-21a.mid
File Written - outfile = None
```

NOTE: The file is actually written to disk and is fine.

And Thank You for creating mido!

**Comment by jnguyen1098** [comment]
Yeah, I think the `filename` field is just for use in the `MidiFile` constructor when opening _existing_ MIDIs. You don't specify `filename` in your example, so the value ends up defaulting to `None` [as shown here](https://github.com/mido/mido/blob/7bc7432d8e6024d800394ad7f1a66d8ffb879abe/mido/midifiles/midifiles.py#L300).

We see in the [`save`](https://github.com/mido/mido/blob/7bc7432d8e6024d800394ad7f1a66d8ffb879abe/mido/midifiles/midifiles.py#L440) method that it just writes to file but doesn't update the actual instance itself, a responsibility that it likely should not have.

**Comment by II-Milou-II** [comment]
For me, this is not an issue.

`self.filename` is stored at `__init__` but never read and never written. So `self.filename` currently only represents the filename that was used when the object was loaded (except when using `file` instead of `filename`, in which case we don't know the filename and `self.filename` is None so this is not exactly representing this accuratly)

For me, the best solution is to consider that `self.filename` has no well-defined meaning and to simply remove it (since it is not used by the library). This is a very small Pull Request that I have done #651.

We can also decide that `self.filename` should represent the last filename we interacted with, as proposed here. But I think that we didn't want this (if we want, I can do a one-line Pull Request to add this as indicated by @jnguyen1098 )

I'll put a second message in a few minutes that explains how to fix the particular issue of @JeffRocchio (I want it in a second message, for me fixing this code is not related with the mido library)

**Comment by II-Milou-II** [comment]
Your code should be :

```
class OutputMIDI:
    def __init__(self, outputFile):
        self.midiOut = mido.MidiFile()
        self.outFilePath = outputFile

    def writeFile(self):
        print(f'Writing File - outfile = {self.outFilePath}')
        self.midiOut.save(self.outFilePath)
        # Just test if the save updates the object's filename param:
        print(f'File Written - outfile = {self.outFilePath}')
```

I think that you want `filename` to be the file that you just saved in but `filename` could reasonably be interpreted in different ways. If we do the fix that you propose, other people might expect a different behaviour.
Since you want a specific behaviour, the mido library should not implement this specific behaviour but you should implement it in your class (as I have done above), this is even simpler

To avoid any further confusion around this unused attribute, I think deleting it, as I proposed in my previous message, is the best solution.
